### PR TITLE
feat(deadm-integration): add cluster id and runbook

### PR DESCRIPTION
**Pull request by boranx**
Fixes: https://issues.redhat.com/browse/OSD-7249

An example created snitch

<div class="summary_item" id="most-recent-incident" style="margin: 0px 0px 12px; padding: 0px 0px 12px; border-width: 0px 0px 1px; border-top-style: initial; border-right-style: initial; border-bottom-style: solid; border-left-style: initial; border-top-color: initial; border-right-color: initial; border-bottom-color: rgb(210, 210, 210); border-left-color: initial; border-image: initial; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: proxima-nova, &quot;Proxima Nova&quot;, helvetica, arial, sans-serif; font-size: 16px; vertical-align: baseline; color: rgb(80, 86, 82); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(243, 245, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="margin: 0px; padding: 0px 0px 6px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 20px; vertical-align: baseline; color: rgb(60, 103, 127);">Most Recent Incident</h3><p style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 16px; vertical-align: baseline;">Everything is looking good!</p></div><div class="summary_item" id="most-recent-check-in" style="margin: 0px 0px 12px; padding: 0px 0px 12px; border-width: 0px 0px 1px; border-top-style: initial; border-right-style: initial; border-bottom-style: solid; border-left-style: initial; border-top-color: initial; border-right-color: initial; border-bottom-color: rgb(210, 210, 210); border-left-color: initial; border-image: initial; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: proxima-nova, &quot;Proxima Nova&quot;, helvetica, arial, sans-serif; font-size: 16px; vertical-align: baseline; color: rgb(80, 86, 82); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(243, 245, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="margin: 0px; padding: 0px 0px 6px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 20px; vertical-align: baseline; color: rgb(60, 103, 127);">Most Recent Check-In</h3>
Last Seen | less than a minute ago
-- | --

<p style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 16px; vertical-align: baseline;">You will be alerted at<span>&nbsp;</span><strong style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: bold; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 16px; vertical-align: baseline;">May 25, 12:00 UTC</strong><span>&nbsp;</span>if your snitch hasn't checked in.</p></div><div class="summary_item" style="margin: 0px 0px 12px; padding: 0px 0px 12px; border-width: 0px 0px 1px; border-top-style: initial; border-right-style: initial; border-bottom-style: solid; border-left-style: initial; border-top-color: initial; border-right-color: initial; border-bottom-color: rgb(210, 210, 210); border-left-color: initial; border-image: initial; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: proxima-nova, &quot;Proxima Nova&quot;, helvetica, arial, sans-serif; font-size: 16px; vertical-align: baseline; color: rgb(80, 86, 82); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(243, 245, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="margin: 0px; padding: 0px 0px 6px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 20px; vertical-align: baseline; color: rgb(60, 103, 127);">Settings</h3>
Type | Basic
-- | --
Token | XXX
URL | https://nosnch.in/XX
Email | XX@nosnch.in
Interval | Hourly

</div><div class="summary_item" style="margin: 0px 0px 12px; padding: 0px 0px 12px; border-width: 0px 0px 1px; border-top-style: initial; border-right-style: initial; border-bottom-style: solid; border-left-style: initial; border-top-color: initial; border-right-color: initial; border-bottom-color: rgb(210, 210, 210); border-left-color: initial; border-image: initial; font-style: normal; font-variant-ligatures: normal; font-variant-caps: normal; font-variant-numeric: inherit; font-variant-east-asian: inherit; font-weight: 400; font-stretch: inherit; line-height: inherit; font-family: proxima-nova, &quot;Proxima Nova&quot;, helvetica, arial, sans-serif; font-size: 16px; vertical-align: baseline; color: rgb(80, 86, 82); letter-spacing: normal; orphans: 2; text-align: start; text-indent: 0px; text-transform: none; white-space: normal; widows: 2; word-spacing: 0px; -webkit-text-stroke-width: 0px; background-color: rgb(243, 245, 244); text-decoration-thickness: initial; text-decoration-style: initial; text-decoration-color: initial;"><h3 style="margin: 0px; padding: 0px 0px 6px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 20px; vertical-align: baseline; color: rgb(60, 103, 127);">Notes</h3><div class="summary-notes" style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: inherit; font-size: 16px; vertical-align: baseline;"><pre style="margin: 0px; padding: 0px; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: &quot;Operator Mono&quot;, &quot;SF Mono&quot;, SFMono-Regular, Consolas, &quot;Liberation Mono&quot;, Menlo, Courier, monospace; font-size: 16px; vertical-align: baseline;"><code style="margin: 0px; padding: 1rem; border: 0px; font-style: inherit; font-variant: inherit; font-weight: inherit; font-stretch: inherit; line-height: inherit; font-family: &quot;Operator Mono&quot;, &quot;SF Mono&quot;, SFMono-Regular, Consolas, &quot;Liberation Mono&quot;, Menlo, Courier, monospace; font-size: 14px; vertical-align: baseline; display: block; outline: none; background-color: rgb(243, 245, 244); overflow-x: auto;">        cluster_id: eab59cda-e2c1-4fbb-87ad-094b67ba5fdd
        runbook: https://github.com/openshift/ops-sop/blob/master/v4/alerts/cluster_has_gone_missing.md</code></pre></div></div>

### CODE REVIEW

File: pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go [hunk 1]
```diff
@@ -331,6 +331,11 @@ func (r *ReconcileDeadmansSnitchIntegration) createSnitch(dmsi *deadmanssnitchv1
 			snitch = snitches[0]
 		} else {
 			newSnitch := dmsclient.NewSnitch(snitchName, dmsi.Spec.Tags, "15_minute", "basic")
+			notes := fmt.Sprintf(`
```
Comment (mmazur) [inline pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go:334]: Does this not produce a needles empty line at the start?

File: pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go [hunk 2]
```diff
@@ -331,6 +331,11 @@ func (r *ReconcileDeadmansSnitchIntegration) createSnitch(dmsi *deadmanssnitchv1
...
+			cluster_id: %s
+			runbook: https://github.com/openshift/ops-sop/blob/master/v4/alerts/cluster_has_gone_missing.md
+			`, cd.Labels["api.openshift.com/id"])
```
Comment (mmazur) [inline pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go:337]: Just to double check – this is the external cluster id, not the internal (hive) id?

File: pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go (hunk 1 above)
Comment (boranx) [reply pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go:334]: if it starts with (`cluster_id.., the formatting goes odd

File: pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go (hunk 2 above)
Comment (boranx) [reply pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go]: nice catch @mmazur !, I've converted it to _externalID_

File: pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go (hunk 1 above)
Comment (mmazur) [reply pkg/controller/deadmanssnitchintegration/deadmanssnitchintegration_controller.go:334]: If it works as it is, then I guess leave it :)

**Pull request merged**